ouroboros-network-framework/src/Ouroboros/Network/ConnectionManager/Core.hs

```diff
@@ -1162,9 +1162,14 @@ withConnectionManager ConnectionManagerArguments {
                 -- If mbTransition is Nothing, it means that the connVar was read
                 -- either in Terminating or TerminatedState. Either case we should
                 -- return Disconnected instead of Connected.
-                return (maybe (Disconnected connId Nothing)
-                              (const $ Connected connId dataFlow handle)
-                              mbTransition)
+                case mbTransition of
```

Comment:
Why is this commit needed? What does it achieve?

Reply:
sorry, which commit?

Reply:
ok, I found it; the last one.

> Why is this commit needed? What does it achieve?

It makes inbound and outbound cases symmetric: make the connection manager responsible for informing the inbound governor about a new connection in both cases.  Should I make the commit message more explicit?

Reply:
No need, but I guess it can't hurt. I just asked because to make sure I understood its scope

Reply:
Fixed: a better commit message is in place.